### ISSUE TITLE
Unable to compile  battery.cpp

### ISSUE THREAD

**Issue by dwarf-king-hreidmar** [Alexays/Waybar#304]
After running meson build successfully with no error output I tried running ninja -C build. I get a bunch of compile errors: 
Fedora 30
Meson output: 
```
 # meson build
The Meson build system
Version: 0.50.1
Source dir: $redacted
Build dir: $redacted
Build type: native build
Project name: waybar
Project version: 0.6.1
Native C compiler: cc (gcc 9.1.1 "cc (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)")
Native C++ compiler: c++ (gcc 9.1.1 "c++ (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Program git found: YES (/usr/bin/git)
Has header "filesystem" : YES
Dependency threads found: YES 
Found pkg-config: /usr/bin/pkg-config (1.6.1)
Dependency libinput found: YES 1.13.1
Found CMake: /usr/bin/cmake (3.14.3)
Dependency fmt found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency fmt

|Executing subproject fmt 
|
|Project name: fmt
|Project version: 5.3.0
|Native C++ compiler: c++ (gcc 9.1.1 "c++ (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)")
|Build targets in project: 1
|Subproject fmt finished.

Dependency fmt from subproject subprojects/fmt found: YES
Dependency wayland-client found: YES 1.17.0
Dependency wayland-cursor found: YES 1.17.0
Dependency wayland-protocols found: YES 1.17
Dependency wlroots found: YES 0.1.0
Dependency gtkmm-3.0 found: YES 3.24.1
Dependency dbusmenu-gtk3-0.4 found: YES 16.04.0
Dependency gio-unix-2.0 found: YES 2.60.1
Dependency jsoncpp found: YES 1.8.4
Dependency sigc++-2.0 found: YES 2.10.1
Dependency libnl-3.0 found: YES 3.4.0
Dependency libnl-genl-3.0 found: YES 3.4.0
Dependency libpulse found: YES 12.2-rebootstrapped
Dependency libudev found: YES 241
Dependency libmpdclient found: YES 2.14
Program wayland-scanner found: YES (/usr/bin/wayland-scanner)
Program gdbus-codegen found: YES (/usr/bin/gdbus-codegen)
Program clang-tidy found: YES (/usr/bin/clang-tidy)
Build targets in project: 4
Found ninja-1.9.0 at /usr/bin/ninja
```

```../src/modules/battery.cpp: In member function ‘void waybar::modules::Battery::getBatteries()’:
../src/modules/battery.cpp:57:27: error: no match for ‘operator/’ (operand types are ‘const std::filesystem::__cxx11::directory_entry’ and ‘const char [9]’)
   57 |           fs::exists(node / "capacity") && fs::exists(node / "uevent") &&
      |                      ~~~~ ^ ~~~~~~~~~~
      |                      |      |
      |                      |      const char [9]
      |                      const std::filesystem::__cxx11::directory_entry
../src/modules/battery.cpp:57:60: error: no match for ‘operator/’ (operand types are ‘const std::filesystem::__cxx11::directory_entry’ and ‘const char [7]’)
   57 |           fs::exists(node / "capacity") && fs::exists(node / "uevent") &&
      |                                                       ~~~~ ^ ~~~~~~~~
      |                                                       |      |
      |                                                       |      const char [7]
      |                                                       const std::filesystem::__cxx11::directory_entry
../src/modules/battery.cpp:58:27: error: no match for ‘operator/’ (operand types are ‘const std::filesystem::__cxx11::directory_entry’ and ‘const char [7]’)
   58 |           fs::exists(node / "status")) {
      |                      ~~~~ ^ ~~~~~~~~
      |                      |      |
      |                      |      const char [7]
      |                      const std::filesystem::__cxx11::directory_entry
../src/modules/battery.cpp:63:27: error: no match for ‘operator/’ (operand types are ‘const std::filesystem::__cxx11::directory_entry’ and ‘const char [7]’)
   63 |           fs::exists(node / "online")) {
      |                      ~~~~ ^ ~~~~~~~~
      |                      |      |
      |                      |      const char [7]
      |                      const std::filesystem::__cxx11::directory_entry
ninja: build stopped: subcommand failed.
```

**Comment by Alexays** [comment]
Can you test by replacing `"capacity"` with `std::string("capacity")`, same for other errors :)

**Comment by dwarf-king-hreidmar** [comment]
Will do later this evening thanks

**Comment by dwarf-king-hreidmar** [comment]
Now i get this error. I don't have much experience with C so maybe added something incorrectly

```
../src/modules/battery.cpp: In member function ‘void waybar::modules::Battery::getBatteries()’:
../src/modules/battery.cpp:57:27: error: no match for ‘operator/’ (operand types are ‘const std::filesystem::__cxx11::directory_entry’ and ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’})
   57 |           fs::exists(node / std::string("capacity")) && fs::exists(node / std::string("uevent")) &&
      |                      ~~~~ ^      ~~~~~~~~~~~~~~~~~~
      |                      |           |
      |                      |           std::string {aka std::__cxx11::basic_string<char>}
      |                      const std::filesystem::__cxx11::directory_entry
In file included from /usr/include/c++/9/valarray:603,
                 from /usr/include/cairomm-1.0/cairomm/context.h:31,
                 from /usr/include/pangomm-1.4/pangomm/context.h:42,
                 from /usr/include/gtkmm-3.0/gtkmm/widget.h:31,
                 from /usr/include/gtkmm-3.0/gtkmm/container.h:28,
                 from /usr/include/gtkmm-3.0/gtkmm/bin.h:27,
                 from /usr/include/gtkmm-3.0/gtkmm/eventbox.h:27,
                 from ../include/ALabel.hpp:4,
                 from ../include/modules/battery.hpp:13,
                 from ../src/modules/battery.cpp:1:
```

**Comment by Alexays** [comment]
I'll test with a fedora image asap

**Comment by Alexays** [comment]
Tested on fedora 30 and that's works, It must be on your side.

**Comment by Alexays** [comment]
Fixed with 80e9ea746bc0d744cd60a7916fa3bdd2e39c8f3e
